Source/WebKit/WebProcess/Network/WebResourceLoader.cpp

```diff
@@ -157,8 +157,10 @@ void WebResourceLoader::didReceiveResponse(ResourceResponse&& response, PrivateR
         response.setDeprecatedNetworkLoadMetrics(Box<NetworkLoadMetrics>::create(WTFMove(*metrics)));
     }
 
-    if (privateRelayed == PrivateRelayed::Yes && mainFrameMainResource() == MainFrameMainResource::Yes)
+    if (privateRelayed == PrivateRelayed::Yes && mainFrameMainResource() == MainFrameMainResource::Yes) {
         WebProcess::singleton().setHadMainFrameMainResourcePrivateRelayed();
+        response.setWasPrivateRelayed(privateRelayed == PrivateRelayed::Yes ? WasPrivateRelayed::Yes : WasPrivateRelayed::No);
```

Comment:
This should not be necessary.  It should've been set in NetworkDataTask::didReceiveResponse